oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/UserPrincipalProvider.java

```diff
@@ -209,6 +203,21 @@ public Iterator<? extends Principal> findPrincipals(int searchType) {
     }
 
     //------------------------------------------------------------< private >---
+
+    GroupPrincipalFactory createGroupPrincipalFactory() {
```

Comment:
missing annotation of return value (adding notnull would be the correct one afaik). 

i am wondering why this method is not private. IMHO nobody outside this class should use it.